ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -113,6 +122,8 @@ public void run() {
         }
       }
     } while (!stop.get());
+
+    ((ExecutorService)cleanerExecutor).shutdown();
```

Comment:
Why not just declare `cleanerExecutor` as and `ExecutorService` to start?

Reply:
Modify it now.